flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -702,11 +704,14 @@ private void disposeAllOperators(boolean logOnlyErrors) throws Exception {
 						operator.dispose();
 					}
 					catch (Exception e) {
-						LOG.error("Error during disposal of stream operator.", e);
+						disposalException = ExceptionUtils.firstOrSuppressed(e, disposalException);
```

Comment:
I think this can cause problems because in `firstOrSuppressed`:
```
public static <T extends Throwable> T firstOrSuppressed(T newException, @Nullable T previous) {
		checkNotNull(newException, "newException");
```

WDYT about changing `firstOrSuppressed` to return any of the passed arguments if the other one is null? (I also had a similar issue :) ).

Reply:
Isn't `e` and thus `newException` always non-null?

Reply:
Yes, you are right.

Reply:
> Isn't `e` and thus `newException` always non-null?

yep, i agree. I think the assumption here is you want to suppress an exception only if you catch a real exception. But I guess no hurt to add the check.

Reply:
So i guess this comment is considered addressed?

Reply:
Yes, please resolve it. Sorry for the confusion.